.github/workflows/createNewVersion.yml

```diff
@@ -84,16 +84,21 @@ jobs:
       - name: Commit new Mobile-Expensify version
         working-directory: Mobile-Expensify
         run: |
-          # Checkout main branch in Mobile-Expensify submodule.
-          # This is IMPORTANT so that when we make changes, we aren't in a detached head state, and when we later push those changes to main, it works!
-          git checkout main
+          # First commit the version changes on detached HEAD
           git add \
             ./Android/AndroidManifest.xml \
             ./app/config/config.json \
             ./iOS/Expensify/Expensify-Info.plist\
             ./iOS/SmartScanExtension/Info.plist \
             ./iOS/NotificationServiceExtension/Info.plist
           git commit -m "Update version to ${{ steps.bumpVersion.outputs.NEW_VERSION }}"
+
+          # Store the commit hash before checkout to ensure we cherry-pick the right commit
+          DETACHED_COMMIT=$(git rev-parse HEAD)
+          
+          # Cherry-pick the version bump commit onto main, to safely handle case where new commits exist on main
+          git checkout main
+          git cherry-pick "$DETACHED_COMMIT"
           if ! git push origin main; then
```

Comment:
Isn't the `git push` going to fail now, because we'll have made a commit on an outdated version of `main`?

Reply:
I think adding `git pull origin main` should resolve that

Reply:
Updated. Let me know if I missunderstood your comment

Reply:
Yes that's what I meant 👍